### PR TITLE
Hostport

### PR DESCRIPTION
**Pull request by shz117**

### CODE REVIEW

File: datanode/client/query_client.go [hunk 1]
```diff
@@ -87,7 +88,12 @@ func (dc *dataNodeQueryClientImpl) QueryRaw(ctx context.Context, host topology.H
 }
 
 func (dc *dataNodeQueryClientImpl) queryRaw(ctx context.Context, host topology.Host, query queryCom.AQLQuery, hll bool) (bs []byte, err error) {
-	url := fmt.Sprintf("http://%s/query/aql?dataonly=1", host.Address())
+	// TODO: @shz remove this hack when we get proper address from host.Address()
+	address := host.Address()
+	if strings.HasPrefix(address, "http://") {
```
Comment (voyager-dw) [inline datanode/client/query_client.go]: better if you see any http:// https://, etc, then no need to prefix protocol
and how about sql query

Comment (shz117) [reply datanode/client/query_client.go]: updated.

sql query will use the same client

File: datanode/client/query_client.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/uber/aresdb/utils"
 	. "io/ioutil"
 	"net/http"
+	url2 "net/url"
```
Comment (jshencode) [inline datanode/client/query_client.go]: url2?

Comment (shz117) [reply datanode/client/query_client.go]: lol ide

**Pull request merged**